### ISSUE TITLE
Data is saved wrongly in parquet file when using s3.to_parquet

### ISSUE THREAD

**Issue by nivf33** [awslabs/aws-data-wrangler#492]
Hi,
I'm using awswrangler 1.10.0, however I've tried to upgrade to the latest version 2.2.0 and it's happening there as well.
I'm trying to use s3.to_parquet with three columns of nested structures types (one for example: [https://github.com/awslabs/aws-data-wrangler/issues/480](url) ).

I'm not sure what is causing that, but I've checked inside the parquet file as well as from Athena and the data seems wrong, including cases of duplicated values inside an array column, as well as other scenarios of wrong values being displayed in arrays and probably taken from other rows.

I've checked and the data seems valid in my flow including as DataFrame, everything seems to get wrong when calling the to_parquet function.

(I've tried to add just as a test the dtype parameter but it didn't help).

The data is a list of dicts of the following type with three nested columns:

`[
  {
    "a": "123",
    "b": "123",
    "c": "test",
    "d": true,
    "e": "2000-02-10T15:03:49.888Z",
    "f": "2000-02-10T13:23:42.496Z",
    "g": [
      {
        "custom_a": "a",
        "custom_b": "b",
        "custom_c": "c"
      },
      {
        "custom_a": "d",
        "custom_b": "e",
        "custom_c": "f"
      }
    ],
    "h": null,
    "i": "2000-02-10",
    "j": [
      {
        "aa": "1",
        "dd": "dfg"
      },
      {
        "bb": "2",
        "ee": "dfg"
      },
      {
        "cc": "3",
        "ff": "dfg"
      }
    ],
    "k": [
      {
        "asd": "123",
        "sdf": "test_a",
        "dfg": "456",
        "fgh": "test_b"
      },
      {
        "asd": "123",
        "sdf": "test_c",
        "dfg": "456",
        "fgh": "test_d"
      }
    ],
    "l": "2000-01-12T05:23:01.417Z",
    "m": "test",
    "n": "test",
    "o": 0,
    "p": null,
    "q": null,
    "r": "test"
  }
]`

**Important note**
I've tried to take one record only and to use the function only on it and it worked, the issue seems to happen when there are many rows in the data.

Thanks

**Comment by maxispeicher** [comment]
Can you maybe share a concrete DataFrame where this issue happens? I'm not able to reproduce it on my side.

Thanks

**Comment by nivf33** [comment]
Hi Max, working on providing you a file for testing
Thanks

**Comment by nivf33** [comment]
Hi Max,
I've started to work on a file to share with you, then I've tried few more scenarios:

- Running without array type columns is valid
- I've reduced the columns to only two (int and array) and the issue exists from the 400th record.
- to_csv works properly, as well as fastparquet which isn't included in wrangler.
- awswrangler to_parquet compression with gzip doesn't work as well.
- The issue reproduce also when using directly pandas to_paruqet without wrangler.

My conclusion is that the issue is related to pyarrow and not directly to wrangler by these tests.

Thanks
